### DISCUSSION TITLE
Nuxt3 `useRuntimeConfig` is not defined.

### DISCUSSION THREAD

**Discussion by itsmnthn** [vuejs/pinia#1285] [Help and Questions]
I am testing pinia store in `nuxt3` app.

Inside `setup()` of store I'm using `useRuntimeConfig` to get initial value for counter from public config variables and I got this error `ReferenceError: useRuntimeConfig is not defined` don't know how to solve it

```Javscript
// store/counter.ts

...
state: () => {
    const runtimeConfig = useRuntimeConfig()
    const count = runtimeConfig.public.count
    return {
      ...
      count
      ...
    }
  },
...
```

code
```Javascript
// store/counter.test.ts

import { fileURLToPath } from 'node:url'
import { describe, expect, it, beforeEach } from 'vitest'
import { setActivePinia, createPinia } from 'pinia'
import { useCounter } from './counter'
import { setup } from '@nuxt/test-utils'

await setup({
  rootDir: fileURLToPath(new URL('../', import.meta.url)),
  server: true,
  browser: true,
})

describe('Counter Store', () => {
  beforeEach(() => {
    // creates a fresh pinia and make it active so it's automatically picked
    // up by any useStore() call without having to pass it to it:
    // `useStore(pinia)`
    setActivePinia(createPinia())
  })

  it('increments', () => {
    const counter = useCounter()
    expect(counter.n).toBe(0)
    counter.increment()
    expect(counter.n).toBe(1)
  })

  it('increments by amount', () => {
    const counter = useCounter()
    counter.increment(10)
    expect(counter.n).toBe(10)
  })
})

```

**Comment by itsmnthn** [comment]
I used `import { useRuntimeConfig } from '../node_modules/nuxt/dist/app' // can use from '#app' but it breaks tests` to solve to `useRuntimeConfig` is not defined but now I get `nuxt` instance unavailable

**Reply by SafwanHak** [reply]
Hi @itsmnthn,  did you get it to work?

**Reply by itsmnthn** [reply]
Instead using runtime configuration I used `process.env.var directly`

**Comment by avivace** [comment]
Is there any update on this? Is there a way to access `useRuntimeConfig` in modules?

**Reply by iusethemouse** [reply]
Hey @avivace, have you had any breakthroughs in terms of this specific issue?

**Reply by avivace** [reply]
@iusethemouse Unfortunately, nope. I'm stuck with `process.env.ENTRY_NAME` ...

**Comment by mimoo** [comment]
this looks cleaner and it works as well:

```
import { useRuntimeConfig } from "#imports";
```

**Reply by ianjamieson** [reply]
What do you have in your `jest.config.ts` set up for `moduleNameMapper`? I am trying various combinations, but all lead to jest not being able to find `useRuntimeConfig` anywhere.

**Comment by ianjamieson** [comment]
I ended up using

```
globalThis.useRuntimeConfig = () => {
  return {
    public: {
       something: 'value'
    },
  };
};
```

To mock `useRuntimeConfig`

**Comment by gbarros1994** [comment]
`  runtimeConfig: {
    public: {
      stripePk: process.env.STRIPE_PK_KEY
    }
  `},``
  
  just call it that way:
`  stripe = Stripe(useRuntimeConfig().public.stripePk)`
